src/openrct2/ride/RideData.h

```diff
@@ -142,16 +142,26 @@ struct UpkeepCostsDescriptor
     uint8_t CostPerStation;
 };
 
+using track_group_t = std::array<bool, TRACK_GROUP_COUNT>;
```

Comment:
I'm not so sure this is the best approach of doing this. This field currently acts as a bitfield to define the group, it would be practical to actually have a table that defines the actual groups and then we can reference the group by index which only requires a single byte for 255 entries.

Reply:
That would require hardcoding or otherwise registering every desired combination of available and extra track groups/elements. I think that will make it more complicated.

Reply:
Also please in the future if you want to do such big refactors create an issue first where we get the chance to discuss it with the team.

Reply:
I created a discussion about it, is that not sufficient?

The ride_track_group_t is not used dynamically unless I'm mistaken about what 'using' does. The array of bools uses a bitfield internally so the memory footprint should be the same before and after this PR.

Reply:
You are confusing specialized std::vector<bool> with std::array<bool, N>. The std::array does not do that. As for the discussion, no one ever greenlit anything. The discussions are there to "discuss" the proposed thing and get to a conclusion before starting the work.